stock_reception_screen_measuring_device/models/stock_reception_screen.py

```diff
@@ -71,29 +95,62 @@ def _compute_package_has_missing_dimensions(self):
             else:
                 record.package_has_missing_dimensions = False
 
-    def measure_current_packaging(self):
-        self.ensure_one()
+    @api.model
+    def _measure_packaging(self, packaging):
         device = self.env["measuring.device"].search(
             [("is_default", "=", True)], limit=1
         )
         if not device:
             error_msg = _("No default device set, please configure one.")
             _logger.error(error_msg)
             self._notify(error_msg)
-            return UserError(error_msg)
+            raise UserError(error_msg)
         if device._is_being_used():
             error_msg = _("Measurement machine already in use.")
             _logger.error(error_msg)
             self._notify(error_msg)
-            return UserError(error_msg)
+            raise UserError(error_msg)
 
-        self.product_packaging_id._measuring_device_assign(device)
+        packaging._measuring_device_assign(device)
         return True
 
+    def measure_current_packaging(self):
+        self.ensure_one()
+        return self._measure_packaging(self.product_packaging_id)
+
+    def _get_smaller_package_without_dimensions_domain(self):
+        self.ensure_one()
+        return [
+            ("product_id", "=", self.current_move_product_id.id),
+            ("qty", "<", self.product_packaging_id.qty),
+            "|",
+            "|",
+            "|",
+            ("packaging_length", "=", 0),
+            ("width", "=", 0),
+            ("height", "=", 0),
+            ("max_weight", "=", 0),
+        ]
+
+    def _get_smaller_package_without_dimensions(self):
+        self.ensure_one()
+        domain = self._get_smaller_package_without_dimensions_domain()
+        return self.env["product.packaging"].search(domain, order="qty desc", limit=1)
```

Comment:
Looks weird, if you want the "smaller" package, shouldn't you order by `qty asc` with the limit set to 1 ?

Reply:
@mmequignon ping

Reply:
no, we measure from the bigger to the smallest
bix box -> medium box -> small box

Reply:
We want the bigger package among the smaller packages missing measurements.